src/main.rs

```diff
@@ -71,7 +74,12 @@ fn apply_pass(pass: &dyn Pass, args: &Args, files: &[PathBuf]) -> Result<String,
         .into_par_iter()
         .map(|file| pass.adapt(args, file))
         .try_fold(String::new, |acc, test_case: Result<_, Error>| {
-            Ok(format!("{}\n{}", acc, test_case?))
+            let test_case = test_case?;
+            if test_case.is_empty() {
+                Ok(acc)
```

Comment:
When returning a proper error instead of a returning and implicit tristate type, this change is unnecessary.